cmd/cluster/agent/create.go

```diff
@@ -29,6 +29,7 @@ func NewCreateCommand(opts *core.CreateOptions) *cobra.Command {
 	cmd.Flags().StringVar(&opts.AgentPlatform.APIServerAddress, "api-server-address", opts.AgentPlatform.APIServerAddress, "The API server address that should be used for components outside the control plane")
 	cmd.Flags().StringVar(&opts.AgentPlatform.AgentNamespace, "agent-namespace", opts.AgentPlatform.AgentNamespace, "The namespace in which to search for Agents")
 	cmd.MarkFlagRequired("agent-namespace")
+	cmd.MarkPersistentFlagRequired("pull-secret")
```

Comment:
why is this needed and it wasn't before?

Reply:
When I removed `cmd.MarkPersistentFlagRequired("pull-secret") from core, so I re-inserted it in each of the refrences except AWS and eventually Azure when support gets added.